src/state/slices/opportunitiesSlice/selectors/lpSelectors.ts

```diff
@@ -47,11 +47,20 @@ export const selectEarnUserLpOpportunity = createDeepEqualOutputSelector(
     if (!opportunityMetadata) return
 
     const [underlyingToken0AmountCryptoBaseUnit, underlyingToken1AmountCryptoBaseUnit] =
-      opportunityMetadata?.underlyingAssetIds.map((assetId, i) =>
+      opportunityMetadata?.underlyingAssetIds.map((underlyingAssetId, i) =>
         bnOrZero(lpAssetBalanceCryptoBaseUnit)
-          .times(opportunityMetadata?.underlyingAssetRatiosBaseUnit[i])
-          .div(bn(10).pow(bnOrZero(assets[assetId]?.precision)))
-          .toFixed(6),
+          // to LP asset base unit
+          .times(
+            fromBaseUnit(
+              opportunityMetadata?.underlyingAssetRatiosBaseUnit[i],
+              assets[underlyingAssetId]?.precision ?? 18,
```

Comment:
let's bail from this whole selector if we don't have an asset rather than opt chaining precision here

Reply:
43b517b9d